source/extensions/common/aws/region_provider_impl.cc

```diff
@@ -134,9 +134,10 @@ absl::optional<std::string> AWSConfigFileRegionProvider::getRegionSet() {
 // https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html
 //
 RegionProviderChain::RegionProviderChain() {
-  add(createEnvironmentRegionProvider());
-  add(createAWSCredentialsFileRegionProvider());
-  add(createAWSConfigFileRegionProvider());
+  // TODO(nbaws): Verify that bypassing virtual dispatch here was intentional
```

Comment:
cc @nbaws

Reply:
This should be OK because `RegionProviderChain` is the most specific implementation class.

Reply:
ty for this. For my benefit, would the root cause fix be to make RegionProviderChain final?